equinox/_module.py

```diff
@@ -305,25 +303,24 @@ def __new__(
         for f in dataclasses.fields(cls):
             if f.name not in cls.__init__.__annotations__:
                 continue  # Odd behaviour, so skip.
+
```

Comment:
Actually, I think I'm noticing a bug in the existing code here -- I think we should only be running all of this if `has_dataclass_init`? Worth fixing up here, if you agree?

Reply:
Done.